common/types/layer.go

```diff
@@ -144,6 +146,15 @@ func BytesToNodeID(buf []byte) (id NodeID) {
 	return id
 }
 
+// ExtractNodeIDFromSig extracts the NodeID from a signature.
+func ExtractNodeIDFromSig(msg, sig []byte) (NodeID, error) {
```

Comment:
doesn't seem intuitive to live here. how about creating a new file nodeid.go and move everything NodeID there?

Reply:
I put it here because the `NodeID` type, its methods and `BytesToNodeID` are also defined here.

I can move all of those to a new file if you want?

Reply:
code moved to `nodeid.go`